packages/opentelemetry-exporter-jaeger/src/jaeger.ts

```diff
@@ -51,6 +51,9 @@ export class JaegerExporter implements SpanExporter {
     localConfig.username = localConfig.username || process.env.JAEGER_USER;
     localConfig.password = localConfig.password || process.env.JAEGER_PASSWORD;
     localConfig.host = localConfig.host || process.env.JAEGER_AGENT_HOST;
+    if (process.env.JAEGER_SERVICE_NAME) {
+      localConfig.serviceName = process.env.JAEGER_SERVICE_NAME;
+    }
```

Comment:
```suggestion
      localConfig.serviceName = localConfig.serviceName || process.env.JAEGER_SERVICE_NAME;
```

Reply:
That was my initial draft. I've been going back and forth on this. The reason I went this way is mostly motivated by my personal use case: I'm using the Jaeger Operator [which automatically injects](https://github.com/jaegertracing/jaeger-operator/blob/master/pkg/inject/sidecar.go#L302) the environment variable `JAEGER_SERVICE_NAME` into containers. This is quire useful because it keeps the service names consistent. In the Java Spring Boot Jaeger implementation, the environment variable also takes precedence over any local configuration. This currently causes my services to inconsistently named.

I definitely understand if you'd prefer to keep the precedence consistent. However I think that host/port/username/password are different from the service name in that the latter is constant so it is unlikely to _not_ be set, whereas for the former parameters one can provide sensible defaults.

I'm OK with either way, making the local config take precedence "just" means that I'll have to move the service name configuration inside my application if I want to set a constant service name and allow it to be overriden per env.

Reply:
Just to clarify @naseemkullah : the serviceName should take precedence over the env var? Despite my minor complaint ;)?

Reply:
:) Hmm, I am going to ask @open-telemetry/javascript-approvers to chime in here as I am unsure now. 🤷‍♂️

Reply:
@open-telemetry/javascript-approvers friendly ping, @chgl would like to add an env var that would take precedence over in code config, differing from the behaviour of the other env vars as per their explanation above, please advise

Reply:
Sorry to take so long to get to this. We now have a proper env var spec which does not contain this environment variable. I think you should open a spec issue with this question so it can be consistent across SIGs.

Reply:
No problem :) 
I realized this is a bit more involved. While `_SERVICE_NAME` is (not yet at least) part of the spec, there's https://github.com/open-telemetry/opentelemetry-specification/issues/709. The `JAEGER_` vars used here aren't spec-compliant either (see https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-environment-variables.md#jaeger-exporter, which requires the `OTEL_JAEGER_`-prefix. see also https://github.com/open-telemetry/opentelemetry-js/issues/1582), instead coming from https://www.jaegertracing.io/docs/1.20/client-features/.

I think it's reasonable to use the `JAEGER_SERVICE_NAME` for now, add support for the `OTEL_JAEGER_`, vars as part of a new PR, and finally support `OTEL_SERVICE_NAME` depending on the outcome of https://github.com/open-telemetry/opentelemetry-specification/issues/709.

The question of precedence still remains open: should the `localConfig` take precedence over env vars? If both are to be supported, should `OTEL_JAEGER_` override `JAEGER_`? Is this something I might raise as an issue in the spec?

Reply:
>The question of precedence still remains open: should the localConfig take precedence over env vars? If both are to be supported, should OTEL_JAEGER_ override JAEGER_? Is this something I might raise as an issue in the spec?

I think this is something the spec should address.